lib/rules/no-native-reassign.js

```diff
@@ -17,6 +18,8 @@ module.exports = {
             recommended: true
         },
 
+        deprecated: true,
```

Comment:
Agree with @platinumazure `deprecated` should be outside of docs, because it's used for other purposes then just documentation. For example we are excluding deprecated rules from `eslint:all` based on this key.